homework/index.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+function main() {
+
+    const mainUrl = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+
+    // initial page: create header and main container.
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('h3', header, { text: 'HYF Repositories' });
+    const mainContainer = createAndAppend('main', root, { class: 'main-container' });
+
+    fetchAndRender(mainUrl);
+
+    async function fetchAndRender(url) {
+
```

Comment:
I would personally not define all of the other functions inside the `main` function. While this helps to keeps those function out of the global scope, I would prefer to use a block scope for that. That would then look like this:

```js
'use strict';

{
  let mainContainer;

  function main() {
    const mainUrl = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';

    // initial page: create header and main container.
    const root = document.getElementById('root');
    const header = createAndAppend('header', root);
    createAndAppend('h3', header, { text: 'HYF Repositories' });
    mainContainer = createAndAppend('main', root, { class: 'main-container' });

    fetchAndRender(mainUrl);
  }

  async function fetchAndRender(url) {
    //...
  }

  // all your other functions

  window.addEventListener('load', main);
}
```

This has the same effect, but now I do not have to hunt around going over your large main function to see where you define functions and where you might have some hidden code that is executed directly. In your case there is no such hidden code, but as a reviewer I had to check it anyway. (It is good to keep an audience in mind when you write code).